### PR TITLE
pylint fstrings and more

### PR DESCRIPTION
**Pull request by marksmayo**
Some more tidy-ups with f-strings, uneeded vars, simplified additions and encodings.

### CODE REVIEW

File: rocketry/_version.py [hunk 1]
```diff
@@ -490,12 +490,12 @@ def render_pep440_old(pieces):
     if pieces["closest-tag"]:
         rendered = pieces["closest-tag"]
         if pieces["distance"] or pieces["dirty"]:
-            rendered += ".post%d" % pieces["distance"]
+            rendered += f".post{pieces['distance']}"
```
Comment (Miksus) [inline rocketry/_version.py:493]: This produces the same (verified manually).

File: rocketry/_version.py [hunk 2]
```diff
@@ -490,12 +490,12 @@ def render_pep440_old(pieces):
...
             if pieces["dirty"]:
                 rendered += ".dev0"
     else:
         # exception #1
-        rendered = "0.post%d" % pieces["distance"]
+        rendered = f"0.post{pieces['distance']}"
```
Comment (Miksus) [inline rocketry/_version.py:498]: This produces the same (verified manually).

File: rocketry/_version.py [hunk 3]
```diff
@@ -570,7 +570,7 @@ def render(pieces, style):
     elif style == "git-describe-long":
         rendered = render_git_describe_long(pieces)
     else:
-        raise ValueError("unknown style '%s'" % style)
+        raise ValueError(f"unknown style '{style}'")
```
Comment (Miksus) [inline rocketry/_version.py:573]: This produces the same (verified manually).

**Review by Miksus — CHANGES_REQUESTED** [review]
In ``versioneer.py`` there is a minor thing that functionally differs from previous (a verbose print statement). This is very minor however.

**Review by Miksus — APPROVED** [review]

**Comment by Miksus** [comment]
Thanks again for the contribution! By the way, how are you using Pylint? I'm thinking of adding some linters (ie. using Flake8) to the CI to keep the code tidy in the future as well.

**Comment by marksmayo** [comment]
I do it in two ways - on the command line (pylint_runner is a handy wrapper for it, and with -d options to hide ones I'm not trying to fix), and in VSCode where you edit the user settings.json file and hide the ones you don't want to see as well.

**Comment by marksmayo** [comment]
You can add it as a Github action, and just -d the ones that currently exist as issues, thus preventing any new ones from being added at least?

**Pull request merged**